Race in TestTaskExecutor.testTaskHandle

This test fails non-deterministically with:
```
java.lang.AssertionError: expected [1] but found [0]
Expected :1
Actual   :0
 <Click to see difference>


	at org.testng.Assert.fail(Assert.java:94)
	at org.testng.Assert.failNotEquals(Assert.java:494)
	at org.testng.Assert.assertEquals(Assert.java:123)
	at org.testng.Assert.assertEquals(Assert.java:370)
	at org.testng.Assert.assertEquals(Assert.java:380)
	at com.facebook.presto.execution.TestTaskExecutor.testTaskHandle(TestTaskExecutor.java:141)
	at sun.reflect.GeneratedMethodAccessor1.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:696)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:882)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1189)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:124)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:343)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:305)
	at org.testng.SuiteRunner.run(SuiteRunner.java:254)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
	at org.testng.TestNG.run(TestNG.java:1057)
	at org.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:74)
	at org.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:121)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
```

It's pretty easy to repro if you set the invocation count to 10k

--- a/presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java
+++ b/presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java
@@ -240,7 +240,7 @@ public class TaskExecutor
                 else {
                     // add this to the work queue for the task
                     taskHandle.enqueueSplit(prioritizedSplitRunner);
-                    // if task is under the limit for gaurenteed splits, start one
+                    // if task is under the limit for guaranteed splits, start one
                     scheduleTaskIfNecessary(taskHandle);
                     // if globally we have more resources, start more
                     addNewEntrants();

--- a/presto-main/src/test/java/com/facebook/presto/execution/TestTaskExecutor.java
+++ b/presto-main/src/test/java/com/facebook/presto/execution/TestTaskExecutor.java
@@ -130,15 +130,20 @@ public class TestTaskExecutor
             beginPhase.register();
             Phaser verificationComplete = new Phaser();
             verificationComplete.register();
-            TestingJob driver = new TestingJob(beginPhase, verificationComplete, 10);
+            TestingJob driver1 = new TestingJob(beginPhase, verificationComplete, 10);
+            TestingJob driver2 = new TestingJob(beginPhase, verificationComplete, 10);
 
             // force enqueue a split
-            taskExecutor.enqueueSplits(taskHandle, true, ImmutableList.of(driver));
+            taskExecutor.enqueueSplits(taskHandle, true, ImmutableList.of(driver1));
             assertEquals(taskHandle.getRunningSplits(), 0);
 
             // normal enqueue a split
-            taskExecutor.enqueueSplits(taskHandle, false, ImmutableList.of(driver));
+            taskExecutor.enqueueSplits(taskHandle, false, ImmutableList.of(driver2));
             assertEquals(taskHandle.getRunningSplits(), 1);
+
+            // let the split continue to run
+            beginPhase.arriveAndDeregister();
+            verificationComplete.arriveAndDeregister();
         }
         finally {
             taskExecutor.stop();

